lib/cxml/sender.rb

```diff
@@ -17,5 +17,19 @@ def render(node)
       end
       node
     end
+
+    private
+
+    # This can return two data types to provide backwards compatibility.
+    # Previous behaviour was always to receive and return one credential, thus
+    # this behaviour is unchanged. We can now also accept and return several
+    # credentials in an array.
+    def generate_credentials(credential_array)
+      credentials = []
+      credential_array.each do |single_credential_hash|
+        credentials << CXML::Credential.new(single_credential_hash)
+      end
+      credentials.count == 1 ? credentials.first : credentials
```

Comment:
What happens here if the credentials.count is less than 1 or greater than 1?  Or will we never reach one of these scenarios?

Reply:
If it's 1 (i.e. the only thing in the array) we just return credential.first, i.e. just the credential without the array wrapper. This is to continue supporting the default behaviour now (which is one credential, no array).

The other side of this conditional is the new feature - if there are multiple credentials, we return the whole array.

If there are zero credentials, we wouldn't make it this far anyway but that is very unlikely since these requests are built by software rather than humans.

Reply:
Ok.  No probs - understood.  Could we also refactor this code to use map eg:

```
credentials = credential_array.map { |single_credential_hash| CXML::Credential.new(single_credential_hash) }.compact
credentials.count == 1 ? credentials.first : credentials
```